File: mobile/lib/dtos/change_password_dto.dart
class ChangePasswordDTO {
  final String currentPassword;
  final String newPassword;
  final String confirmPassword;

  ChangePasswordDTO(
      {required this.currentPassword,
      required this.newPassword,
      required this.confirmPassword});

  Map<String, dynamic> toJson() => {
        'currentPassword': currentPassword,
        'newPassword': newPassword,
        'confirmPassword': confirmPassword,
      };
}

File: backend/src/main/java/br/unitins/facelocus/dto/ChangePasswordDTO.java
package br.unitins.facelocus.dto;

import br.unitins.facelocus.constraintvalidator.PasswordsEqualConstraint;
import jakarta.validation.constraints.NotBlank;

@PasswordsEqualConstraint
public record ChangePasswordDTO(@NotBlank(message = "Informe a senha atual") String currentPassword,
                                @NotBlank(message = "Informe a nova senha") String newPassword,
                                @NotBlank(message = "Confirme a nova senha") String confirmPassword) {
}
